File: testdata/jennies/builders/properties/PythonBuilder/builders/properties.py
import typing
from ..cog import builder as cogbuilder
from ..models import properties


class SomeStruct(cogbuilder.Builder[properties.SomeStruct]):
    _internal: properties.SomeStruct
    __some_builder_property: str = ""

    def __init__(self):
        self._internal = properties.SomeStruct()

    def build(self) -> properties.SomeStruct:
        """
        Builds the object.
        """
        return self._internal    
    
    def id(self, id_val: int) -> typing.Self:    
        self._internal.id_val = id_val
    
        return self
    

File: testdata/jennies/builders/properties/RustBuilder/src/builders/properties.rs
use crate::cog;
use crate::types::properties::SomeStruct;

#[derive(Debug, Clone)]
pub struct SomeStructBuilder {
    internal: SomeStruct,
    errors: Vec<cog::BuildError>,
}

impl SomeStructBuilder {
    pub fn new() -> Self {
        Self {
            internal: SomeStruct::default(),
            errors: Vec::new(),
        }
    }
}

impl Default for SomeStructBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl SomeStructBuilder {
    pub fn id(mut self, id: i64) -> Self {
        self.internal.id = id;

        self
    }
}

impl cog::Builder<SomeStruct> for SomeStructBuilder {
    fn build(&self) -> Result<SomeStruct, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(self.internal.clone())
    }
}
